### PR TITLE
Adding null check for nextSibling

### PR DESCRIPTION
**Pull request by BryceEWatson**
## Description

I've been debugging the below stack trace, and this is a simple fix that prevents it. I'm not 100% sure if there are other considerations here so please let me know, but this removes the exception when nextSibling is null.

```
Uncaught TypeError: Cannot read property 'nextSibling' of null
    at Component.___forEachNode (Component-91f50b8e.js:541)
    at getNodes (Component-91f50b8e.js:142)
    at Component.destroy (Component-91f50b8e.js:269)
    at detachedNodes.forEach (index-b9133752.js:809)
    at Array.forEach (<anonymous>)
    at morphdom (index-b9133752.js:801)
    at updateManager.___batchUpdate (Component-91f50b8e.js:515)
    at Object.batchUpdate [as ___batchUpdate] (update-manager-2ee5a85d.js:63)
    at Component.___rerender (Component-91f50b8e.js:498)
    at Component.update (Component-91f50b8e.js:451)
```
## Motivation and Context

I've seen this error reported in various contexts, but for me it is happening when I update state and remove a parameter from state. The removed parameter causes a UI element to be removed from the page, and it then proceeds to loop through the siblings of this component until nextSibling is null, at which point it will throw the exception. After the exception is thrown no further state updates are working.

A few issues I've seen this reported are https://github.com/marko-js/marko/issues/920 and https://github.com/marko-js/marko/issues/1051

This change fixes the above error by avoiding a null reference error, regardless of the root cause.

## Checklist:

- [x] My code follows the code style of this project.
- [NA] I have updated/added documentation affected by my changes.
- [x] I have read the **CONTRIBUTING** document.
- [NA] I have added tests to cover my changes.
- [x] All new and existing tests passed.

_Disclaimer: Contributions via GitHub pull requests are gladly accepted from their original author. Along with any pull requests, please state that the contribution is your original work and that you license the work to the project under the project's open source license. Whether or not you state this explicitly, by submitting any copyrighted material via pull request, email, or other means you agree to license the material under the project's open source license and warrant that you have the legal authority to do so._

### CODE REVIEW

**Comment by cameronbraid** [comment]
I had done this myself for a while, however what its doing is treating the symptom not the cause and I suspect the PR will be rejected.

**Comment by BryceEWatson** [comment]
Yes I agree it's just treating the symptom, but if there's no negative impact to doing so.. why not? I hope the cause is still fixed though, but this will at least let us continue working.

**Comment by scttdavs** [comment]
If the next sibling being null is expected in some cases, then it may be ok
to do this and throw in a debug logging statement. But if not, then I think
the best thing it can do is fail early and loudly. Swallowing the error may
give it the appearance of it being fine, but there may be a fault somewhere
that is not discovered until later, and it will be impossible to debug
without any errors or stack traces.

**Comment by BryceEWatson** [comment]
Agreed, this most likely isn't expected to be null and failing loudly is nice. If anything this PR can serve as a quick hack for anyone googling in the future.

Some additional context in case it helps identify the root cause:
It seems like the component being removed has `___endNode` assigned to a `<!--$marko-->` comment. It never runs across this comment and instead hits the end of the siblings.

Inspecting the html it looks like the comment is inserted at the bottom of the component, along with an extra closing `</div>` tag. This is where I hit a dead end, and someone with more marko knowledge would need to weigh in I think.

TLDR; Component.___endNode is incorrectly assigned to a marko comment, and is inserted oddly / incorrectly in the dom.

**Comment by BryceEWatson** [comment]
Adding some observations:
Flow 1 - Error does occur during this flow:
1. Page loads with element visible on the page (state exists for it)
2. State update causes a re-render of the element, but it still exists.
3. State update causes the removal of the element, hits the original error.
Flow 2 - Error does *not* occur during this flow:
1. Page loads with element visible.
2. State update causes removal of element, this works fine.

Difference in call stacks:
Flow 1 (Errors):
See stack above, calls `Component.destroy` on all detached nodes.
Flow 2 (No error):
```
___forEachNode (Component-91f50b8e.js:541)
___detach (Component-91f50b8e.js:532)
morphChildren (index-7b0079e3.js:330)
insertVirtualNodeBefore (index-7b0079e3.js:79)
morphChildren (index-7b0079e3.js:441)
morphComponent (index-7b0079e3.js:156)
morphChildren (index-7b0079e3.js:347)
morphdom (index-7b0079e3.js:800)
updateManager.___batchUpdate (Component-91f50b8e.js:515)
batchUpdate (update-manager-2ee5a85d.js:63)
___rerender (Component-91f50b8e.js:498)
update (Component-91f50b8e.js:451)
updateComponents (update-manager-2ee5a85d.js:44)
updateUnbatchedComponents (update-manager-2ee5a85d.js:16)
(anonymous) (nextTick-browser-40824fd0.js:18)
postMessage (async)
setImmediate (nextTick-browser-40824fd0.js:27)
scheduleUpdates (update-manager-2ee5a85d.js:35)
queueComponentUpdate (update-manager-2ee5a85d.js:101)
___queueUpdate (Component-91f50b8e.js:417)
___set (State-legacy-623e5b28.js:91)
___replace (State-legacy-623e5b28.js:61)
replaceState (Component-91f50b8e.js:364)
```

**Comment by cameronbraid** [comment]
@BryceEWatson FYI I have implemented a test case that produces this error https://github.com/marko-js/marko/pull/1064

**Comment by BryceEWatson** [comment]
Great! Closing this PR since we dont understand the root cause + we have a test case now.
